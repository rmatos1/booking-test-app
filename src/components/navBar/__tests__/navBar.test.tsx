import { describe, test } from '@jest/globals';
import { fireEvent, render, waitFor } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import { NavBar } from '..';

const setup = (): JSX.Element => {
  return (
    <MemoryRouter>
      <NavBar />
    </MemoryRouter>
  );
};

describe('<NavBar />', () => {
  test('should render correctly', () => {
    const wrapper = render(setup());

    expect(wrapper).toMatchSnapshot();
  });

  test('link should become active on click', async () => {
    const wrapper = render(setup());

    const myBookingsLink = wrapper.getByText('My Bookings');

    fireEvent.click(myBookingsLink);

    await waitFor(() => {
      expect(myBookingsLink.classList.contains('active')).toBe(true);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
