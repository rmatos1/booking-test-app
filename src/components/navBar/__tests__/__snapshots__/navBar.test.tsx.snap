// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> link should become active on click 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: unset;"
  >
    <div>
      <nav
        class="sc-bdfCDU cJXGYm"
      >
        <a
          class="sc-gsTDqH cIkufZ"
          href="/"
        >
          New Booking
        </a>
        <a
          aria-current="page"
          class="sc-gsTDqH zgixn active"
          href="/my-bookings"
        >
          My Bookings
        </a>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="sc-bdfCDU cJXGYm"
    >
      <a
        class="sc-gsTDqH cIkufZ"
        href="/"
      >
        New Booking
      </a>
      <a
        aria-current="page"
        class="sc-gsTDqH zgixn active"
        href="/my-bookings"
      >
        My Bookings
      </a>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`<NavBar /> should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: unset;"
  >
    <div>
      <nav
        class="sc-bdfCDU cJXGYm"
      >
        <a
          aria-current="page"
          class="sc-gsTDqH zgixn active"
          href="/"
        >
          New Booking
        </a>
        <a
          class="sc-gsTDqH cIkufZ"
          href="/my-bookings"
        >
          My Bookings
        </a>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="sc-bdfCDU cJXGYm"
    >
      <a
        aria-current="page"
        class="sc-gsTDqH zgixn active"
        href="/"
      >
        New Booking
      </a>
      <a
        class="sc-gsTDqH cIkufZ"
        href="/my-bookings"
      >
        My Bookings
      </a>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
